src/core/exec-invoke.c

```diff
@@ -526,8 +523,16 @@ static int setup_output(
                         return fileno;
 
                 /* Duplicate from stdout if possible */
-                if (can_inherit_stderr_from_stdout(context, o, e))
+                if (can_inherit_stderr_from_stdout(context, o, e)) {
+                        r = fd_is_writable(STDOUT_FILENO);
+                        if (r < 0)
+                                return r;
```

Comment:
Rather than fail, maybe better to gracefully fallback to /dev/null ?
```c
if (r <= 0) {
        if (r < 0)
                log_warning_errno(...);
        else
                log_warning(...);
        return open_null_as(...);
}